lib/apiservers/engine/backends/container.go

```diff
@@ -671,13 +673,13 @@ func (c *Container) ContainerAttach(prefixOrName string, ca *backend.ContainerAt
 	vc := cache.ContainerCache().GetContainer(prefixOrName)
 
 	if vc == nil {
-		//FIXME: If we didn't find in the cache, we should goto the port layer and
-		//see if it exists there.  API server might have been bounced.  For now,
-		//just return error
-
 		return derr.NewRequestNotFoundError(fmt.Errorf("No such container: %s", prefixOrName))
 	}
 
+	if running, err := IsContainerRunning(vc.ContainerID); !running {
```

Comment:
What does this do in the case of `attach` followed by `start` as is seen with run and start -ai ?